app/models/results/web_response_parser.rb

```diff
@@ -99,5 +107,17 @@ def form_items_in_mission
       @form_items_in_mission ||=
         FormItem.where(mission_id: @response.mission_id).pluck(:id).index_by(&:itself)
     end
+
+    def options_in_mission?(option_node_ids)
+      option_node_ids.all? { |id| option_in_mission?(id) }
+    end
+
+    def option_in_mission?(option_node_id)
+      option_node_id.blank? || option_nodes_in_mission[option_node_id].present?
+    end
+
+    def option_nodes_in_mission
+      @option_nodes_in_mission ||= OptionNode.where(mission: @response.mission).pluck(:id).index_by(&:itself)
+    end
```

Comment:
These 3 methods are duplicated now. It seems like they should go in `concerns/`, yeah? Not sure what to name it, the concept of "fooable" has never been very clear to me. Advice appreciated.

Reply:
May be moot if we switch to using a similar approach here, which is to use a scope when looking up the option ID. See line 163 in Answer.